sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_client_connection.py

```diff
@@ -2539,6 +2539,32 @@ def __GetBodiesFromQueryResult(result):
         # Query operations will use ReadEndpoint even though it uses POST(for regular query operations)
         request_params = _request_object.RequestObject(typ, documents._OperationType.SqlQuery)
         req_headers = base.GetHeaders(self, initial_headers, "post", path, id_, typ, options, partition_key_range_id)
+
+        #check if query has prefix partition key
+        isPrefixPartitionQuery = kwargs.pop("isPrefixPartitionQuery", None)
+        if isPrefixPartitionQuery:
+            #here get the overlap, then do one of two scenarios
+            partition_key_definition = kwargs.pop("partitionKeyDefinition", None)
+            pkproperties = partition_key_definition
+            partition_key_definition = PartitionKey(path=pkproperties["paths"], kind=pkproperties["kind"])
+            partition_key_value = pkproperties["partition_key"]
+            feedrangeEPK = partition_key_definition.get_epk_range_for_prefix_partition_key(partition_key_value)
+            over_lapping_ranges = self._routing_map_provider.get_overlapping_ranges(id_, [feedrangeEPK])
+            if over_lapping_ranges:
```

Comment:
Here I see the cases for if there is exactly 1 partition and <1 partition. Both of with are handled correctly. However, I do not see the case for if there is more than one partition in the range. Do you have the split range case handled here? Ideally, I think that case should be represented here. In .NET, we first check the length of the overlapping ranges list and if it is greater than 1, we return a `410:1002` (PartitionKeyRangeGone) that bubbles it up to a higher layer so deal with the split.

Reply:
The routing map provider that has already been implemented in the python sdk specifically deals with the case of split partitions so we won't end in the situation of overlapping ranges being over 1 at this point.  There are also test available for it in the python SDK that also specifically test that case.

Reply:
What will the routing map provider do? Will it throw an exception that the query layer needs to handle? In that case, would the query retry? But how does the retry work if the input (partial PK) is not changing?

Reply:
Resolved in the last iteration